src/network/protocol/messages/merkleblock.zig

```diff
@@ -0,0 +1,146 @@
+const std = @import("std");
+const protocol = @import("../lib.zig");
+
+const Sha256 = std.crypto.hash.sha2.Sha256;
+const BlockHeader = protocol.BlockHeader;
+const CompactSizeUint = @import("bitcoin-primitives").types.CompatSizeUint;
+
+/// MerkleBlockMessage represents the "MerkleBlock" message
+///
+/// https://developer.bitcoin.org/reference/p2p_networking.html#merkleblock
+pub const MerkleBlockMessage = struct {
+    block_header: BlockHeader,
+    transaction_count: u32,
+    hash_count: CompactSizeUint,
```

Comment:
you can remove this field as it represent the same information as `self.hashes.len`, just init a CompactSizeUint form it when required